src/Compilers/CSharp/Portable/Parser/LanguageParser.cs

```diff
@@ -3185,6 +3185,12 @@ private bool IsPossibleAccessorModifier()
             // to the user that this is not allowed.
             if (IsPossibleModifier())
             {
+                if (this.CurrentToken.Kind == SyntaxKind.RefKeyword)
+                {
+                    // Although a legal modifier, it is never parsed unless before a struct definition
+                    return false;
+                }
+
                 var peekIndex = 1;
                 while (IsPossibleModifier(this.PeekToken(peekIndex)))
```

Comment:
>IsPossibleModifier(this.PeekToken(peekIndex))) [](start = 23, length = 46)

Should we have a special handling for a ref modifier here?

Reply:
I don't think so. We will eat it here, and later produce error messages. I'll add some test cases for that.

---
In reply to: [158534670](https://github.com/dotnet/roslyn/pull/23905#discussion_r158534670) [](ancestors = 158534670)